src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1394,10 +1394,21 @@ bool SQLStorage::loadInstalledVersions(const std::string& ecu_serial, boost::opt
   return true;
 }
 
-bool SQLStorage::hasPendingInstall() {
+bool SQLStorage::hasPendingInstall(bool ecu_only) {
```

Comment:
if an ECU serial is a parameter of this method with a default value EcuSerial::Unknown() or an empty string then the sql query can be simpler SELECT count(*) FROM installed_versions where is_pending = 1 AND serial = %serial%